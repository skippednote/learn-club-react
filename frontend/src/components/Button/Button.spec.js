import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Index as Button } from './index';

describe('Button', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <Button apperence="primary" className="increment">
        Button
      </Button>
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('call the onClick handler when pressed', () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <Button apperence="primary" className="increment" onClick={onClick}>
        Button
      </Button>
    );
    wrapper.simulate('click');
    expect(onClick).toHaveBeenCalled();
  });
});
