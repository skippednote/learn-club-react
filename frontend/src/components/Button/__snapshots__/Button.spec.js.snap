// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<Button
  apperence="primary"
  className="increment"
>
  Button
</Button>
`;
